core/src/main/groovy/com/novoda/gradle/release/AndroidArtifacts.groovy

```diff
@@ -1,41 +1,48 @@
 package com.novoda.gradle.release
 
 import org.gradle.api.Project
+import org.gradle.api.Task
 import org.gradle.api.tasks.bundling.Jar
 import org.gradle.api.tasks.javadoc.Javadoc
 
-class AndroidArtifacts implements Artifacts {
+class AndroidArtifacts {
 
-    def all(String publicationName, Project project) {
-        [sourcesJar(publicationName, project), javadocJar(publicationName, project), mainJar(project)]
+    private Project project
+
+    AndroidArtifacts(Project project) {
+        this.project = project
+    }
+
+    def all(String publicationName) {
+        [sourcesJar(publicationName), javadocJar(publicationName), mainJar()]
     }
 
-    def sourcesJar(String publicationName, Project project) {
+    def sourcesJar(String publicationName) {
         project.task(publicationName + 'AndroidSourcesJar', type: Jar) {
             classifier = 'sources'
             from project.android.sourceSets.main.java.srcDirs
         }
     }
 
-    def javadocJar(String publicationName, Project project) {
-        def androidJavadocs = project.task(publicationName + 'AndroidJavadocs', type: Javadoc) {
+    def javadocJar(String publicationName) {
+        Task androidJavadocs = project.task(publicationName + 'AndroidJavadocs', type: Javadoc) {
             source = project.android.sourceSets.main.java.srcDirs
-            classpath += project.files(project.android.getBootClasspath().join(File.pathSeparator))
+            classpath += project.android.bootClasspath
```

Comment:
I hope it does. ;)
I couldn't really test it. Would I have to deploy to a local repo for that?
Can't you just link it to some sample similar to what we do in the command-plugin or the sqlite-analyzer?